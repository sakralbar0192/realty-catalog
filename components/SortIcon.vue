<template>
  <div :class="styles['sort-icon']" :data-test="`sort-icon-${direction}`">
    <!-- Chevron Up -->
    <svg
      :class="[styles['sort-icon__chevron'], styles['sort-icon__chevron--up'], { [styles.active]: direction === 'asc' }]"
      xmlns="http://www.w3.org/2000/svg"
      width="7"
      height="4"
      fill="none"
      viewBox="0 0 7 4"
    >
      <g fill-rule="evenodd" clip-rule="evenodd">
        <path
          fill="currentColor"
          d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"
        />
      </g>
    </svg>

    <!-- Chevron Down -->
    <svg
      :class="[styles['sort-icon__chevron'], styles['sort-icon__chevron--down'], { [styles.active]: direction === 'desc' }]"
      xmlns="http://www.w3.org/2000/svg"
      width="7"
      height="4"
      fill="none"
      viewBox="0 0 7 4"
    >
      <g fill-rule="evenodd" clip-rule="evenodd">
        <path
          fill="currentColor"
          d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"
        />
      </g>
    </svg>
  </div>
</template>

<script setup lang="ts">
import styles from '~/assets/styles/components/SortIcon.module.scss'

interface Props {
  direction?: 'asc' | 'desc' | null
}

defineProps<Props>()

defineOptions({
  name: 'SortIcon'
})
</script>