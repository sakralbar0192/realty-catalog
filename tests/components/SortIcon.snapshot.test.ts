import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import SortIcon from '~/components/SortIcon.vue'

describe('SortIcon Snapshots', () => {
  it('should match snapshot when no direction', () => {
    const wrapper = mount(SortIcon)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when direction is asc', () => {
    const wrapper = mount(SortIcon, {
      props: {
        direction: 'asc',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when direction is desc', () => {
    const wrapper = mount(SortIcon, {
      props: {
        direction: 'desc',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when direction is null', () => {
    const wrapper = mount(SortIcon, {
      props: {
        direction: null,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
