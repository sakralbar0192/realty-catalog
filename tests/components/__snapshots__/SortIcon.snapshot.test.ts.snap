// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortIcon Snapshots > should match snapshot when direction is asc 1`] = `
"<div class="_sort-icon_224acd" data-test="sort-icon-asc">
  <!-- Chevron Up --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--up_224acd _active_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg><!-- Chevron Down --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--down_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg>
</div>"
`;

exports[`SortIcon Snapshots > should match snapshot when direction is desc 1`] = `
"<div class="_sort-icon_224acd" data-test="sort-icon-desc">
  <!-- Chevron Up --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--up_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg><!-- Chevron Down --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--down_224acd _active_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg>
</div>"
`;

exports[`SortIcon Snapshots > should match snapshot when direction is null 1`] = `
"<div class="_sort-icon_224acd" data-test="sort-icon-null">
  <!-- Chevron Up --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--up_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg><!-- Chevron Down --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--down_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg>
</div>"
`;

exports[`SortIcon Snapshots > should match snapshot when no direction 1`] = `
"<div class="_sort-icon_224acd" data-test="sort-icon-undefined">
  <!-- Chevron Up --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--up_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg><!-- Chevron Down --><svg class="_sort-icon__chevron_224acd _sort-icon__chevron--down_224acd" xmlns="http://www.w3.org/2000/svg" width="7" height="4" fill="none" viewBox="0 0 7 4">
    <g fill-rule="evenodd" clip-rule="evenodd">
      <path fill="currentColor" d="M4.034.253A.846.846 0 0 0 3.428 0c-.229 0-.444.09-.605.253L.168 3.013a.583.583 0 0 0 0 .817.566.566 0 0 0 .807 0l2.453-2.555L5.882 3.83a.566.566 0 0 0 .808 0 .583.583 0 0 0 0-.817L4.034.253Z"></path>
    </g>
  </svg>
</div>"
`;
